### ISSUE TITLE
Add useful whisper features

### ISSUE THREAD

**Issue by thewh1teagle** [k2-fsa/sherpa-onnx#1125]
Ordered by importance

- Start and stop timestamps for each segment
- new segment decoded callback (or I would like to know better how to stream decode efficiently as whisper has 30s limit)
- progress? in case of long decode
- Word timestamps
- temperature
- initial prompt

**Comment by hlevring** [comment]
Yes, it would be great to implement whisper word timestamps using dtw
